### ISSUE TITLE
Option to not-clobber existing secret, opting to patch it instead. 

### ISSUE THREAD

**Issue by avo-sepp** [bitnami-labs/sealed-secrets#1205]
**Which component**:
Controller

**Is your feature request related to a problem? Please describe.**
Yes, my problem is that I have a case where I need to use Kustomize to PATCH an exiting Kubernetes secret for ArgoCD. ArgoCD has other values in this secret that it populates at runtime. I have Kustomize and the patches saved in Git, but I cannot commit the secret to Git because it is in plaintext. 

I thought I could use Kustomize to instead apply a sealed-secret. However, sealed-secret's default is to clobber/overwrite any existing secret. 

**Describe the solution you'd like**
An annotation/label where I could tell the sealed-secret controller to use a PATCH operation and NOT overwrite any existing secret with the same name. 

**Describe alternatives you've considered**
Alternatives I've considered store the secret outside of Git and apply it with some SIG tool or script.

**Comment by alemorcuq** [comment]
It's an interesting suggesion, @avo-sepp. We will look into it.

**Comment by alemorcuq** [comment]
Hey, @avo-sepp. Could you show us a concrete example of what you have in mind to help us understand this issue? I understand you want to Sealed Secrets to partially update an already existing secret.

**Comment by tvanderka** [comment]
I recently had the same use-case. I want to create a secret with sealedsecret from git, and later have additional keys added by other controllers/operators at runtime. I also do need an option to keep the secret when sealedsecret is deleted.

Possible solution:
sealedsecrets.bitnami.com/managed: "false" -> do NOT recreate&own the existing secret, only update/patch keys from ss
(skip if no "managed" annotation)

For an example look at https://github.com/argoproj/argo-cd/blob/master/docs/operator-manual/argocd-secret.yaml, there are a few "autogenerated" keys I could leave undefined, use sealedsecret only for webhook.github.secret, or add additional accounts.* later. This is currently not possible as ss overwrites it completely.

**Comment by avo-sepp** [comment]
I would like to add another, related, but different option for `sealedsecrets.bitnami.com/clobber: "false"` which would modify the patch operation such that it does not overwrite a value that already exists. In this case the end result of valueB would be `valueB: "bcd"` instead of `"123"`. Which might be a nice addition to this PATCH functionality.

**Comment by agarcia-oss** [comment]
Hi @avo-sepp @tvanderka Thanks for your feedback and examples! We're going to include the `patch` proposal in the project backlog. 
However, regarding the latest `clobber` flag we believe that could break the design philosophy of the sealed secrets. Once you use a sealed secret, it should be assumed that the sealed secret contains the source of truth of the affected secret. Can't you get the same result by simply removing from your Sealed Secrets the data elements you don't want to manage with Sealed secrets and apply the `patch` flag? 

Example Sealed secret
```
data:
  valueC: "456"
```
Resulting sealed secret
```
data:
  valueA: "abc"
  valueB: "bcd"
  valueC: "456"
```

**Comment by avo-sepp** [comment]
I agree the clobber idea would break philosophy. I rescind that suggestion. 

Thank you for adding the patch proposal to the backlog!

**Comment by alemorcuq** [comment]
Hey @avo-sepp. We have merged #1259 adding the `sealedsecrets.bitnami.com/patch: "true"` annotation. It can be used alone or in combination with the `managed` annotation. The only difference is that when combined with the `managed` annotation the SealedSecret will take ownership of the Secret, so that when the SealedSecret is removed the Secret is also removed.

For your use case, I believe you'd like to use it alone. I'll close this ticket now, but feel free to reopen it or open a new one if you have any questions.
